jukebox/src/main/webapp/js/script.js

```diff
@@ -1,25 +1,112 @@
-function generateGroup() {
+/**
+This file contains all database reads/writes as well some minor webpage
+modifications. It's most likely not secure to have the database available
+on the client-side, so this set-up is temporary (although will likely remain
+as apart of the MVP)
+ */
 
-    //send to server once database schema decided
-    var groupName = document.getElementById('group_name').value;
-    window.location.href = '/create_pin.html';
+//Setting up Firebase configuration
+var firebaseConfig = {
+
+    apiKey: "AIzaSyDkEJE1taOS3R_dxakJjMB98I-3Apjuhxk",
+    authDomain: "jukebox-sps.firebaseapp.com",
+    databaseURL: "https://jukebox-sps.firebaseio.com",
+    projectId: "jukebox-sps",
+    storageBucket: "jukebox-sps.appspot.com",
+    messagingSenderId: "729977109157",
+    appId: "1:729977109157:web:0f47881639cc145866ab4c",
+    measurementId: "G-9M791HTW5T"
+}
+
+// Initialize Firebase
+firebase.initializeApp(firebaseConfig);
+var db = firebase.firestore();
+
+//This function sets up the room page, getting the room name based on PIN
+function presentRoom(isHost) {
+
+    const queryParameters = new URLSearchParams(window.location.search);
+    const roomRef = db.collection('rooms').doc(queryParameters.get('pin'))
+    //If the a room with the PIN exists, then load its given components
+    roomRef.get().then(function(roomSnapshot) {
+        if (roomSnapshot.exists) {
+            var groupName = roomSnapshot.data().name 
+            //Distinguish between host and general user
+            if (isHost) { 
+                document.getElementById('group_name').textContent = roomSnapshot.data().name + " (Host)";
+            } else {
+                document.getElementById('group_name').textContent = roomSnapshot.data().name + " (General)";
+            }
+        } else {
+            window.location.href = "/pin_error.html"
+        }
+    })
+    document.getElementById("overlay_pin").textContent = queryParameters.get("pin")
 }
 
+//Displays pin on create_pin.html
 function presentPIN() {
-    
-    //Eventually put this on java serverside code once database schema decided
-    var alphaNumeric = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ';
-    var PIN = '';
-    for(var i = 0; i < 6; i++) {
-        PIN += alphaNumeric[Math.floor(Math.random() * alphaNumeric.length)];
-    }
-    document.getElementById('PIN').textContent = PIN;
+
+    const queryParameters = new URLSearchParams(window.location.search);
+    document.getElementById('PIN').textContent = queryParameters.get('pin');
```

Comment:
maybe we can rename "PIN" to something like "presented-pin" to differentiate it from what we get from the URL?